dask/distributed/tests/test_distributed.py

```diff
@@ -46,31 +47,40 @@ def teardown_module():
             break
         time.sleep(0.1)
 
-
-def test_dask_client_from_ipclient():
-    c = Client()
-    dask_client = dask_client_from_ipclient(c)
-
-    # data
-    a = np.arange(100).reshape(10, 10)
-    d = da.from_array(a, ((5, 5), (5, 5)))
-
+@contextmanager
+def ipcluster():
+    setup_cluster()
     try:
-        # test array.mean
-        expected = a.mean(axis=0)
-        d1 = d.mean(axis=0)
-        result = d1.compute(get=dask_client.get)
-        assert_array_almost_equal(result, expected)
-
-        # test ghosting
-        d2 = da.ghost.ghost(d, depth=1, boundary='reflect')
-        d3 = da.ghost.trim_internal(d2, {0: 1, 1: 1})
-        result1 = d3.compute(get=dask_client.get)
-        assert_array_almost_equal(result1, a)
-
+        c = Client()
+        yield c
     finally:
-        # close the workers
-        dask_client.close(close_scheduler=True)
+        teardown_cluster()
+
+@pytest.mark.slow
+def test_dask_client_from_ipclient():
+    with ipcluster() as c:
+        dask_client = dask_client_from_ipclient(c)
+
+        # data
+        a = np.arange(100).reshape(10, 10)
+        d = da.from_array(a, ((5, 5), (5, 5)))
+
+        try:
+            # test array.mean
+            expected = a.mean(axis=0)
+            d1 = d.mean(axis=0)
+            result = d1.compute(get=dask_client.get)
+            assert_array_almost_equal(result, expected)
+
+            # test ghosting
+            d2 = da.ghost.ghost(d, depth=1, boundary='reflect')
+            d3 = da.ghost.trim_internal(d2, {0: 1, 1: 1})
+            result1 = d3.compute(get=dask_client.get)
+            assert_array_almost_equal(result1, a)
+
+        finally:
+            # close the workers
+            dask_client.close(close_scheduler=True)
 
 
 def test_gh_248():
```

Comment:
This test is supposed to be run with the distributed scheduler, but it doesn't look like that was happening before.

Out of scope: we should have a context manager which sets up a local distributed cluster using multiprocessing (not ipython) to run tests like this.